python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -189,6 +191,11 @@ def get_partition_keys_in_range(
                 {list(key for key in keys_exist if keys_exist[key] is False)}"""
             )
 
+        # in the simple case, simply return the single key in the range
+        if partition_key_range.start == partition_key_range.end:
+            return [cast(T_str, partition_key_range.start)]
+
+        partition_keys = self.get_partition_keys(dynamic_partitions_store=dynamic_partitions_store)
```

Comment:
can we add a comment here detailing why we defer calling `get_partition_keys`? Could see someone refactoring and then undo-ing the changes here

Reply:
good call, added !